src/WooCommerceCheckout.php

```diff
@@ -28,6 +28,36 @@ public static function init() {
       add_filter('woocommerce_checkout_fields', __CLASS__ . '::addConfirmationEmailCheckoutField');
       add_action('woocommerce_checkout_process', __CLASS__ . '::checkConfirmationEmailField');
     }
+
+    add_filter( 'woocommerce_form_field', __CLASS__ . '::woocommerceFormField', 10, 4 );
+  }
+
+  /**
+   * Adds inline error messages to checkout fields when needed.
+   *
+   * @implements woocommerce_form_field
+   *
+   * @return string
+   */
+  public static function woocommerceFormField($field, $key, $args, $value) {
+    if (strpos($field, '</label>') === FALSE) {
+      return $field;
+    }
+
+    $error = [];
+    if ($args['required'] ?? FALSE) {
+      $errorMessage = sprintf(__('%s is a required field.', 'woocommerce'), $args['label']);
+      $error[]      = '<span class="error-required" style="display:none">' . $errorMessage . '</span>';
```

Comment:
Variables should not be aligned. Imagine bigger multi line arrays holding keys sorted by name. If you add another, longer key, it's possible you will have to adjust dozens of lines as based on this single addition. 
```suggestion
      $error[] = '<span class="error-required" style="display:none">' . $error_message . '</span>';
```